index.js

```diff
@@ -393,6 +395,7 @@ function session(options) {
         debug('saving %s', this.id);
         savedHash = hash(this);
         _save.apply(this, arguments);
+        debug('session saved');
```

Comment:
This prints before the session is saved, as the saving operation is async.

Reply:
This is based on I am assuming that this line is desired to be similar to the destroy logs, with the behavior of "destroying.." being logged before the store call and then "... destroyed" called when the store returns back to this module without an error.

Reply:
thanks @dougwilson . yes, the desire is to follow `destroy`. the intent is simple - many issues in this repo are about session not being saved or inability to know about it.

thinking about it more, I can achieve this at the `save` site only via structural changes to the `save` function. that is, anchoring the `save` call with a custom callback, check for errors, or else print the debug message and then invoke the original callback, if any.

Before implementing it and submitting for your review, I want to pro-actively check with you on it - will you be open for such a change?

Reply:
Hi @gireeshpunathil yes, that sounds good to me, to just mimic the destroy way in the save area.

Reply:
@dougwilson - done, pls have a look. Here is a sample output from a simple program with debug on, for your reference:
```
  express-session fetching qs3pPenGDQy0hOXWCw_G8lD_ZLJw5AA1 +0ms
  express-session no session found +5ms
  express-session session generate +1ms
  express-session saving 7G28RiMV20yaNOzU3UeC5_qJMKek7X6f +4ms
  express-session set-cookie connect.sid=s%XXX; Path=/; Expires=YYY; HttpOnly +4ms
  express-session session saved +3ms
```